gensim/models/phrases.py

```diff
@@ -133,19 +133,30 @@ def __init__(self, sentences=None, min_count=5, threshold=10.0,
         `delimiter` is the glue character used to join collocation tokens, and
         should be a byte string (e.g. b'_').
 
+        `recode_to_utf8` is an optional parameter to tokenize the input words into
+        utf8 encoding. For faster Phrases, you may use `recode_to_utf8=False` if you have
+        larger RAM for this memory-time tradeoff.
+
         """
         if min_count <= 0:
-            raise ValueError("min_count should be at least 1")
+            min_count = 1
+            warnings.warn("min_count should be at least 1. Continuing with min_count=1")
 
         if threshold <= 0:
             raise ValueError("threshold should be positive")
 
+        self.recode_to_utf8 = recode_to_utf8
         self.min_count = min_count
         self.threshold = threshold
         self.max_vocab_size = max_vocab_size
         self.vocab = defaultdict(int)  # mapping between utf8 token => its count
         self.min_reduce = 1  # ignore any tokens with count smaller than this
         self.delimiter = delimiter
+
+        if not recode_to_utf8:
+            if not isinstance(next(iter(next(iter(sentences)))), bytes):
```

Comment:
@menshikh-iv Thanks. I think it would be better to add a test for non-repeatable corpus too, wdyt ? As of now, all the tests are passing already. cc @piskvorky